### PR TITLE
Python: Refactor integration tests (#7698)

### PR DESCRIPTION
**Pull request by Fokko**
* Python: Refactor integration tests

This splits out running and building the integration tests that enables quick development iterations. I've also added ipython that will give a more meaningful error when something goes wrong with the provisioning of the tests.

* Simplify SQL

### CODE REVIEW

**Comment by Fokko** [comment]
Reviewed and approved earlier by @danielcweeks https://github.com/apache/iceberg/pull/7698 But now the caching issues are resolved.

File: python/dev/Dockerfile [hunk 1]
```diff
@@ -37,7 +37,7 @@ RUN mkdir -p ${HADOOP_HOME} && mkdir -p ${SPARK_HOME} && mkdir -p /home/iceberg/
 WORKDIR ${SPARK_HOME}
 
 ENV SPARK_VERSION=3.3.2
-ENV ICEBERG_VERSION=1.2.0
+ENV ICEBERG_VERSION=1.2.1
```
Comment (rdblue) [inline python/dev/Dockerfile:40]: 1.3 now?

Comment (Fokko) [reply python/dev/Dockerfile:40]: Yes!

Comment (Fokko) [reply python/dev/Dockerfile:40]:
```suggestion
ENV ICEBERG_VERSION=1.3.0
```

**Comment by Fokko** [comment]
Thanks for the review @rdblue !

**Pull request merged**